### ISSUE TITLE
GoRouter migration

### ISSUE THREAD

**Issue by TomBursch** [thunder-app/thunder#250]
**Is your feature request related to a problem? Please describe.**
From the looks of it, you wanted/want to use GoRouter. If this is still the case, I want to create an issue to discuss how to discuss this, see how I or others might be able to help. There are a few things I would change if you want to go down this route.
- Replace `PageView` with shell routes
  - Implement a custom transition for these using the animations packages  
- Move `ThunderBloc` above the Material app (possibly merge with `ThemeBloc` since from the looks of it both store user preferences)
- Get rid of nested scaffolds and have only one in the `ThunderApp`
  - This is easier if you use CustomScrollViews with SliverAppBars
- The community page does multiple things currently (i.e. displaying the feed or a community). I would split this up into community page and feed page. The community page could then be a custom route and be pushed when a community is selected in the drawer, and the feed page would be part of the shell route.

A bit unrelated to this:
- With these changes, you could also move the settings to an action/icon button on the profile tab. I don't think a whole tab is needed for that.
- These changes would allow the drawer to open over the appbar which IMO looks a bit better,

**Additional context**
I've implemented some of the stuff rudimentary, take a look at this branch https://github.com/TomBursch/thunder/tree/go_router for inspiration of how this could look.

If this is not something you want, feel free to close this issue.

**Comment by hjiangsu** [comment]
Yeah, I'm thinking that migrating to `go_router` would be beneficial in the long term. Thanks for creating this issue and thanks for all the suggestions! I think all of those suggestions make sense and they seem like the right direction to go with this

Another potential benefit of using `go_router` is that it'll allow us to structure our routes to handle deep linking more easily in the future

I'll check out the changes you've made and see how we can approach this moving forward!

**Comment by hjiangsu** [comment]
I just took a quick look at the changes and it seems fairly good overall! Would you like to create a draft PR for this, and then I can help work on any conflicts that may have popped up from all the recent changes?

If you would like to make a PR, base it off the `develop` branch!

**Comment by hjiangsu** [comment]
Just a quick update on this - we're no longer relying on `go_router` for navigation to simplify our navigation stack. Because of this, I'll go ahead and close this issue to clean up the backlog. For context: https://github.com/thunder-app/thunder/pull/1613

P.S: Awesome work on KitchenOwl!
